sklearn/discriminant_analysis.py

```diff
@@ -174,6 +175,26 @@ def _class_cov(X, y, priors, shrinkage=None, covariance_estimator=None):
     return cov
 
 
+def _from_discriminant_score_to_probability(discriminant_scores):
```

Comment:
I think that we could go further:

```python
class DiscriminantAnalysisPredictionMixin:
    def decision_function(self, X):
        scores = self._decision_function(X)
        if len(self.classes_) == 2:
            return scores[:, 1] - scores[:, 0]
        return scores

    def predict(self, X):
        scores = self._decision_function(X)
        return self.classes_.take(scores.argmax(axis=1))

    def predict_proba(self, X):
        scores = self._decision_function(X)
        likelihood = np.exp(scores - scores.max(axis=1)[:, np.newaxis])
        return likelihood / likelihood.sum(axis=1)[:, np.newaxis]

    def predict_log_proba(self, X):
        return np.log(self.predict_proba(X))
```

Then, we have the choice to define the docstring in the mixin and to be generic enough such that it is true in both classes (QDA and NearestCentroid) or define the docstring in the class and call `super`.

We will need for sure to redefine `predict` in `NearestCentroid` to call directly `pairwise_distance_argmin` in the case this is possible (when the prior is uniform) and otherwise call the generic implementation.

Reply:
I would keep this in the `discriminant_analysis.py`